CMakeLists.txt

```diff
@@ -14,24 +14,22 @@ endif()
 
 # Imath version
 
-project(Imath VERSION 3.0.2 LANGUAGES C CXX)
+project(Imath VERSION 3.1.0 LANGUAGES C CXX)
 
 set(IMATH_VERSION_EXTRA "dev" CACHE STRING "Extra version tag string for Imath build")
 
 # See https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
-set(IMATH_SOVERSION 28)
+set(IMATH_SOVERSION 29)
 set(IMATH_SOREVISION 0) 
 set(IMATH_SOAGE 0) 
 set(IMATH_LIB_VERSION "${IMATH_SOVERSION}.${IMATH_SOREVISION}.${IMATH_SOAGE}")
 
-set(IMATH_VERSION_MAJOR ${CMAKE_PROJECT_VERSION_MAJOR})
-set(IMATH_VERSION_MINOR ${CMAKE_PROJECT_VERSION_MINOR})
-set(IMATH_VERSION_PATCH ${CMAKE_PROJECT_VERSION_PATCH})
-set(IMATH_VERSION ${CMAKE_PROJECT_VERSION})
 if (NOT IMATH_VERSION_EXTRA STREQUAL "")
-  set(IMATH_VERSION "${CMAKE_PROJECT_VERSION}-${IMATH_VERSION_EXTRA}")
+  set(IMATH_VERSION "${Imath_VERSION}-${IMATH_VERSION_EXTRA}")
```

Comment:
This turns out not to be ok. IMATH_VERSION cascades down to getting put in the exported ImathConfigVersion.cmake as 

    set(PACKAGE_VERSION "3.1.0-dev")

and that's in fact an invalid cmake package version, which has to be of the form MAJOR.MINOR[.PATCH[.TWEAK]]

We don't want the "-dev" to show up in this variable.

In OIIO, I use the dev designator as a totally separate variable:

    set (PROJECT_VERSION_RELEASE_TYPE "dev")   # "dev", "betaX", "RCY", ""

and then it can be directly tested in cmake code, it gets folded into the string that prints for versions, but not into the OpenImageIO_VERSION variable or the exported configs.

Reply:
Current master Imath's exported config appears to be broken for this reason. (But luckily, 3.0.3 is not. It's mildly broken for a totally different reason. :-))

Reply:
We've had a rough patch of bad luck in the last few days that isn't really anybody's fault, but I think there are lessons that we can draw from it to make things smoother in the future. One of those lessons is that even the minor patch releases should not have changes to the build scripts added mere hours before a tag is published. All patches should go into the RB-x.y branch and sit for 48 hours without any bugs found (including by downstream projects' CI), and only then should the tag (and NO other changes) be added and release made. (Of course, in a TRUE emergency release, where the prior one is simply too broken to be made any worse even if there are mistakes.)

Reply:
I'm aware that I'm the one who urged the "-dev" stuff, and I also didn't spot this problem during the PR reviews. I didn't realize the problem until it had already been merged and trickled downstream. Sorry about that.

Reply:
I'll try proposing a fix for this, since I feel like I'm the one who set us off in the wrong direction.

Reply:
I don't think we have any reason to do anything special or fancy or non-standard with Imath or OpenEXR, it should be just basic version specification/identification stuff, so parroting exactly what other projects do seems like the best approach. I took inspiration for the original cleanup from OIIO but obviously didn't replicated it properly.

Reply:
See #146